sentry/src/main/java/io/sentry/SentryWrapper.java

```diff
@@ -0,0 +1,34 @@
+package io.sentry;
+
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+public final class SentryWrapper {
+  public static <U> Callable<U> wrapCallable(Callable<U> callable) {
+    final IHub oldState = Sentry.getCurrentHub();
+    final IHub newHub = Sentry.getCurrentHub().clone();
+
+    return () -> {
+      Sentry.setCurrentHub(newHub);
+      try {
+        return callable.call();
+      } finally {
+        Sentry.setCurrentHub(oldState);
+      }
+    };
+  }
+
+  public static <U> Supplier<U> wrapSupplier(Supplier<U> supplier) {
```

Comment:
could you add `final` keywords and `@Nullable` or `@NotNull` annotations?
Also, if `supplier` could be null, could you add a null check?